includes/REST_API/Stories_Base_Controller.php

```diff
@@ -58,8 +68,151 @@ class Stories_Base_Controller extends WP_REST_Posts_Controller {
 	 */
 	public function __construct( $post_type ) {
 		parent::__construct( $post_type );
-		$this->namespace = 'web-stories/v1';
-		$this->decoder   = new Decoder();
+		$this->namespace   = 'web-stories/v1';
+		$this->decoder     = new Decoder();
+		$this->experiments = new Experiments();
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @see register_rest_route()
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		parent::register_routes();
+
+		if ( ! $this->experiments->is_experiment_enabled( 'enablePostLocking' ) ) {
+			return;
+		}
```

Comment:
Anything speaking against registering the REST API endpoints all the tie, regardless of whether the feature is enabled? Avoids having to pass in `Experiments` and then stripping it out again once we stabilize the feature.

Reply:
I don't see the harm in this. It is only 7 lines of code. What is the benefit of making an api that developers can use even when we are not finished with it.